### PR TITLE
Data layer

### PR DESCRIPTION
**Pull request by tduffy000**
Hi all, take a look at this initial version of the data layer which will be responsible for pulling the data from the csv file using the d3 library. From there, you all should be able to consume the data into the map & bar-chart using some aggregations. Ideally, the Map and slider should update the React `state` such that the page is dynamically reloaded & the data filtered when the `state` changes. This is still a work-in-progress but putting up for transparency & feedback.

### CODE REVIEW

**Comment by msarker000** [comment]
Sorry for my late comments.

i looked at code. it looks very good. can you  modify your aggregate so that it includes state wise species summary also. bcz i need to show state wise 5 largest species in bar char.

{
ALABAMA: {
    total: 122170,
    species: {
               A: 1000,
               B: 2340,
              C: 3201
             },
       }
ALASKA:{
     total: 34813,
     species: {
            A: 1000,
            B: 2340,
            C: 3201
       },
   }
....
}

File: app/src/DataHandler.js [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+  filterByFish(data) {
+    return this.state.fish === "all" ? data : data.filter(d => d['AFS Name'] === this.state.fish)
+  };
+
+  filterNaN = (record) => {
+    return !isNaN(Number(record[1]))
+  }
+
+  // DATA AGGREGATIONS
+  // TODO: some of the state names are messy
```
Comment (msarker000) [inline app/src/DataHandler.js]: if some state names are messy then you  process landing.csv using python and upload it

Comment (tduffy000) [reply app/src/DataHandler.js]: yeah I'll clean it up, we're expecting them to be all caps like the example in your first comment?

**Comment by tduffy000** [comment]
Will add the state-wise top-n aggregation by another field (e.g. `fish`) as requested by @msarker000 :+1:

**Comment by carlosezmz** [comment]
your code looks awesome, is it possible we can add the initials of each State to the file as well? Because my the way the map is set up is that whenever you click on a state it yields the state initials.

**Comment by tduffy000** [comment]
State mapping addressed in https://github.com/tunghoangt/DataVisualization/pull/4/commits/3ed821f2afdd0636f3b6c0217504671304a6f7ed, I will merge this and work on stitching all the pieces together in another PR.

**Pull request merged**